src/filesystem.ts

```diff
@@ -149,18 +150,14 @@ export class Filesystem {
 
     node.size = size;
     node.offset = this.offset.toString();
-    node.integrity = await getFileIntegrity(p);
+    node.integrity = await getFileIntegrity(streamGenerator());
```

Comment:
Hm... this made me realize that we compute integrity of the original file and not the transformed one? Is that right?